### ISSUE TITLE
Allow support for actions mapped to all devices

### ISSUE THREAD

**Issue by Nibodhika** [godotengine/godot#17942]
**Current status:**
Most single player games need to have several mappings to the same action, or implement a script to map the same button/axis to all controllers, because the user might have more than one controller plugged (or his one controller might have been detected as device 1 for some reason).

**Proposed status:**
Ideally the user should be able to choose when creating an action to map to all controllers instead of a given one. I've tried locally two fixes, but wanted to ask the community about which implementation feels the best and if this is a feature that should be merged into mainstream Godot.

*Special Device Number:*
First and simplest solution is to allow a special value for ```device``` when creating an action (such as -1 or 999) that means all devices, I have done this and tested it and it works as expected. However the interface for adding new actions becomes non-intuitive. Ideally the user should choose in a combo box whether to map to a single device or multiple devices. But I couldn't find any examples in the code of places in the interface that change the editor content depending on a selection, if someone could point me to any such example that would be great.

*Store all devices boolean:*
The other solution (which I like less) is to add a boolean value to the ```InputEvent```, the reason I like this approach less is because it creates an extra variable that's not needed there for any event, and because it changes the structure class, which in turn implies in several changes all around the code. I started to work on this just to familiarize myself more with the structure of this part of the code in Godot, but wasn't able to make the changes propagate to the in-game action (i.e my Project editor would store correctly the value for my boolean, but in-game that boolean was set wrong, also closing and opening the project would reset the value), probably because I didn't wrote any ```IO``` for that variable (because I have no idea where that should be).

Personally I feel this is a good addition to the engine, but there might be some way to do this which I couldn't find in any documentation. I also like the first approach better, and would gladly fork and open a PR if this is deemed a good feature, but would need some help with the interface. Thanks.

**Comment by vnen** [comment]
Cc @Hinsbart

**Comment by Nibodhika** [comment]
Ok, I've managed to get a hold on how the GUI works, very intuitive I must add.

I've created a branch on my personal fork which implements this feature the way I feel is the best, both in code and in UI.

The only issues I have with it is just that I don't like where I declared the ```ALL_DEVICES``` variable (And if it is supposed to be global then probably rename ```DeviceMode::ALL_DEVICES``` to something like ```DeviceMode::ALL_DEVICES_MODE``` to distinguish one from the other).

Also I've introduced some new strings and would be willing to translate them, but I think that should be done separately, right?

Finally I have only tested this on Linux, and although I didn't wrote any platform specific code I would feel more comfortable if other people tried it in different platforms.
